autotest/gcore/vrt_read.py

```diff
@@ -1463,6 +1463,23 @@ def test_vrt_protocol():
     ds = gdal.Open("vrt://data/float32.tif?ot=Int32")
     assert ds.GetRasterBand(1).DataType == gdal.GDT_Int32
 
+    ## 5-element GCP
+    ds = gdal.Open("vrt://data/float32.tif?gcp=0,0,0,10,0")
+    assert ds.GetGCPCount() == 1
+
+    ## multiple GCPs
+    ds = gdal.Open("vrt://data/float32.tif?gcp=0,0,0,10,0&gcp=20,0,10,10,0")
+    assert ds.GetGCPCount() == 2
+
+    ## 4-element GCP also ok
+    ds = gdal.Open("vrt://data/float32.tif?gcp=0,0,0,10&gcp=20,0,10,10")
+    assert ds.GetGCPCount() == 2
+
+    try:
+        ds = gdal.Open("vrt://data/float32.tif?gcp=invalid")
+    except SyntaxError:
+        pass
```

Comment:
```suggestion
    ds = gdal.Open("vrt://data/float32.tif?gcp=invalid")
    assert ds is None
```

Reply:
I don't think exceptions are enabled here, and if I turn them on I get a `RuntimeError` here.

Reply:
ah, thank you - wasn't sure about this at all

Reply:
why doesn't it error on open, though? that's weird to silently return null ... isn't it?

Reply:
See https://gdal.org/api/python_gotchas.html#python-bindings-do-not-raise-exceptions-unless-you-explicitly-call-useexceptions

Reply:
ah nice, thank you 🙏